cmd/kubeadm/app/phases/upgrade/postupgrade.go

```diff
@@ -103,8 +104,14 @@ func PerformPostUpgradeTasks(client clientset.Interface, cfg *kubeadmapi.InitCon
 		errs = append(errs, err)
 	}
 
-	if err := proxy.EnsureProxyAddon(&cfg.ClusterConfiguration, &cfg.LocalAPIEndpoint, client); err != nil {
-		errs = append(errs, err)
+	// Upgrade kube-proxy only if the component config is present.
+	// TODO: consider removing this check once upgrade phases are supported.
+	if cfg.ComponentConfigs.KubeProxy == nil {
```

Comment:
I don't quite understand the logic here.
But I guess it should be `if cfg.ComponentConfigs.KubeProxy != nil {`.
Right ?

Reply:
i will update it.
this particular nil check is viable, to my understanding. it can be nil and a panic will happen.